fedimint-server/src/consensus/engine.rs

```diff
@@ -231,7 +233,7 @@ impl ConsensusEngine {
 
     pub async fn run_session(
         &self,
-        connections: ReconnectPeerConnections<Message>,
+        connections: PeerConnections<Message>,
         session_index: u64,
     ) -> anyhow::Result<()> {
```

Comment:
Now I get it and I really like that change, I'm just still grumpy we somehow need to be able to clone connections and potentially introduce bugs that way through parallel readers. Did you explore splitting sender and receiver side? There's primitives for that in tokio, but only for io/byte streams. We might be able to borrow that design though.

Reply:
Since we are only ever running on session there are no parallel readers. Spitting does not seem to make sense as aleph but expects one network struct.

Reply:
Well, let's hope only people with your level of care will ever touch the relevant code parts :grimacing: